lib/std/Thread/Mutex.zig

```diff
@@ -60,22 +66,50 @@ else if (builtin.os.tag.isDarwin())
 else
     FutexImpl;
 
+const DebugImpl = struct {
+    locking_thread: ?Thread.Id = null,
+    impl: ReleaseImpl = .{},
+
+    inline fn tryLock(self: *@This()) bool {
+        const locking = self.impl.tryLock();
+        if (locking) {
+            self.locking_thread = Thread.getCurrentId();
+        }
+        return locking;
+    }
+
+    inline fn lock(self: *@This()) void {
+        const current_id = Thread.getCurrentId();
+        if (self.locking_thread == current_id) {
```

Comment:
This is technically a data race as thread-1 could be reading `locking_thread` while thread-2 could be writing to it. If possible, `locking_thread` should be put behind an `Atomic()` and use `load/store(Unordered)`. 

Unordered over Monotonic as modification order doesn't need to be ensured; This is a check on the local thread and doesn't need to observe other thread stores. Just needs to be atomic to avoid a data-race.

Reply:
Oh, right. I didn't think about that.

I tried to implement this, however it's failing for `x86`:
```
lib/std/atomic/Atomic.zig:80:37: error: expected 32-bit integer type or smaller; found 64-bit integer type
                else => @atomicLoad(T, &self.value, ordering),
                                    ^
```
Apparently 32 bit systems cannot handle 64-bit atomics.

This might be a problem of Thread.zig, as `Thread.Id` is always 64 bits, while at least the windows and linux implementation appear to only have a 32 bit id.

Do you have any suggestions what I should do? Should I just use a second mutex to lock the `locking_thread`? Or should I put this on ice until `Thread.Id` properly reflects the OS type.